### PR TITLE
Simplying the CMakeLists and enforcing .NET projects are built via CMake

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves the changes requested in https://github.com/novelrt/NovelRT/pull/278

### CODE REVIEW

File: samples/NovelRT.DotNet.Sample/CMakeLists.txt [hunk 1]
```diff
@@ -1,52 +1,53 @@
...
 )
 
-set(DOTNET_SAMPLE_OPTIONS
+set(OPTIONS
   /nologo
-  /p:IntermediateOutputPath=${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/Dotnet_Sample_Build.dir/
-  /p:MSBuildProjectExtensionsPath=${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/Dotnet_Sample_Build.dir/
-  /p:OutDir=${CMAKE_CURRENT_BINARY_DIR}/
-  /p:PublishDir=${CMAKE_CURRENT_BINARY_DIR}/publish/
+  /p:IntermediateOutputPath=${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/Dotnet_Sample_Build.obj/
```
Comment (tannergooding) [inline samples/NovelRT.DotNet.Sample/CMakeLists.txt:12]: Explicitly calling out that I renamed this from `.dir` to `.obj` and make `${CMAKE_CURRENT_BINARY_DIR}` the publish folder.

The binary outputs are now going to a `.bin` folder in the cmake files directory.

File: samples/NovelRT.DotNet.Sample/NovelRT.DotNet.Sample.csproj [hunk 2]
```diff
@@ -31,4 +22,9 @@
     <Message Importance="High" Text="@(ResolvedFileToPublish->'$(PublishDir)%(RelativePath)')" />
   </Target>
 
+    <!-- DO NOT REMOVE: Used to ensure the project is only built via CMake -->
```
Comment (FiniteReality) [inline samples/NovelRT.DotNet.Sample/NovelRT.DotNet.Sample.csproj:25]: Maybe we should export these to a Directory.Build.targets :thinking:

**Review by FiniteReality — APPROVED** [review]
Looks good to me

File: samples/NovelRT.DotNet.Sample/NovelRT.DotNet.Sample.csproj (hunk 2 above)
Comment (tannergooding) [reply samples/NovelRT.DotNet.Sample/NovelRT.DotNet.Sample.csproj:25]: I can do that when I add the tests project. Will also let us normalize some other settings and block automatic import of any other Dir.Build stuff in the file tree

File: samples/NovelRT.DotNet.Sample/CMakeLists.txt (hunk 1 above)
Comment (tannergooding) [reply samples/NovelRT.DotNet.Sample/CMakeLists.txt]: This was updated to be `.dir/bin` and `.dir/obj` in the latest iteration.

File: samples/CAPI/CMakeLists.txt [hunk 3]
```diff
@@ -1,12 +1,12 @@
 find_package(lua 5.3 REQUIRED)
 
-set(CAPI_SOURCES
+set(SOURCES
```
Comment (RubyNova) [inline samples/CAPI/CMakeLists.txt:3]: So I'm curious - what's with the shared `SOURCES` CMake variable and others that you seem to re-use?

Comment (FiniteReality) [reply samples/CAPI/CMakeLists.txt:3]: Each CMakeLists is ran in its own scope; this is just simplifying the names where having a prefix was unnecessary

Comment (RubyNova) [reply samples/CAPI/CMakeLists.txt:3]: Fair enough!

Comment (tannergooding) [reply samples/CAPI/CMakeLists.txt:3]: This is what @FiniteReality called out on the other PR (https://github.com/novelrt/NovelRT/pull/278#discussion_r602920760). 

> a lot of names could likely be simplified, as the CMake set() command only operates within the current scope (unless PARENT_SCOPE is specified), and each CMakeLists is in its own scope IIRC.

This means we can just keep it simple as `SOURCES` and it won't conflict across the distinct CMakeLists.txt

File: src/NovelRT.Interop/CMakeLists.txt [hunk 4]
```diff
@@ -1,10 +1,10 @@
-set(INTEROP_SOURCES
+set(SOURCES
   NrtDebugService.cpp
   NrtInteropErrorHandling.cpp
   NrtLoggingService.cpp
   NrtNovelRunner.cpp
   NrtTransform.cpp
-  
+
```
Comment (RubyNova) [inline src/NovelRT.Interop/CMakeLists.txt:7]: Uhhhh, what changed here?

Comment (FiniteReality) [reply src/NovelRT.Interop/CMakeLists.txt:7]: Removal of whitespace

Comment (tannergooding) [reply src/NovelRT.Interop/CMakeLists.txt:7]: Whitespace was removed automatically by the formatter.

This was a line that was `  \n` (two spaces and a newline), it is now just `\n` (just a newline)

**Pull request merged**